crates/net/network/src/transactions.rs

```diff
@@ -590,6 +590,24 @@ where
                 }
                 tx.send(res).ok();
             }
+            TransactionsCommand::TransactionsHandle { peer_id, requested_hashes, result } => {
+                if let Some(peer) = self.peers.get(&peer_id) {
+                    match peer
+                        .request_tx
+                        .send(RequestMessage::GetTransactions(requested_hashes))
+                        .await
+                    {
+                        Ok(transactions) => {
+                            let _ = result.send(Ok(RequestResult::Fulfilled(transactions)));
+                        }
+                        Err(e) => {
+                            let _ = result.send(Err(e));
+                        }
+                    }
+                } else {
+                    let _ = result.send(Err(RecvError::new("Peer not found")));
+                }
```

Comment:
this should return `peer.request_tx.clone()`

and all the request logic should happen in the context of the `TransactionHandle` where we can use await